apis/k8ssandra/v1alpha1/cassandraconfig_types.go

```diff
@@ -863,6 +863,10 @@ type CassandraYaml struct {
 	// +optional
 	SstablePreemptiveOpenIntervalInMb *int `json:"sstable_preemptive_open_interval_in_mb,omitempty"`
 
+	// Exists in 3.11
+	//+optional
+	StartRpc *bool `json:"start_rpc,omitempty"`
```

Comment:
I would have liked that this is not exposed in the CRD, but maybe that's not possible due to how we're creating the JSON?

Reply:
Agreed, I'd rather see this added internally to the json blurb that we feed into config-builder:

https://github.com/k8ssandra/k8ssandra-operator/blob/21332d9b6060cd1ea3159e5d1f1f664a1264126a/pkg/cassandra/config.go#L58

Reply:
You'd probably need to marshal `c.CassandraYaml` to `map[string]interface{}` first, then add the `start_rpc=false` entry.

Reply:
Some customers are still using Thrift (yes, they should migrate off, but legacy applications are a thing). 

Is there some reason why we wouldn't want these customers to be able to migrate to k8ssandra? It doesn't seem to do us any harm to include an option to enable it.

Reply:
NB: it is definitely technically possible to remove it from the CRD. Not a problem at all.

Reply:
>Some customers are still using Thrift (yes, they should migrate off, but legacy applications are a thing).

That's a fair point. I suspect though that those customers are probably not running in Kubernetes. Supporting legacy features increases the maintenance burden on us with a not so great risk/reward ratio.

Reply:
> I suspect though that those customers are probably not running in Kubernetes.

I suspect the same, but neither of us have done a survey or have any data, so I don't think we should block their migration path based on our subjective mental image of a Thrift user. ([Image](https://twitter.com/QuinnyPig/status/1227135904839417857/photo/1) of likely Thrift user).

Maintenance: what maintenance would be required?  Worth noting that we are an open source project and take PRs. Anyone still using Thrift is welcome to contribute as they find things that are broken, I'd think?

Reply:
@Miles-Garnsey, I was the one pushing to disable thrift altogether. While some users may still be using thrift (they have all my empathy), since it was removed in 4.0 and we only support Cassandra since 3.11, I don't see a lot of value in exposing the ability to start the RPC port.
We could reconsider this if we got a wave of folks wanting to use K8ssandra and Thrift, but at this stage I don't think we should spend time on this.
While maintenance could be minimal for such a thing, anything more in the code means tech debt. Removing this will also allow us to expose one less port in our pods and services (which should be a follow up ticket to this one).

Reply:
I'm going to second what @adejanovski is saying here. We're in the position as a project that we're purposefully opinionated with what we provide. I think the overwhelming opinion of the community as a whole is that we're moving on from thrift. So we can take that opportunity here to simplify. 

It's worth noting that making the decision to omit the option here doesn't mean we couldn't add it later if we saw a strong demand for it, highly unlikely I'd think, but we've got that door. If we put it in now, we can't really unopen that box so to speak. 

Let's take the simpler path, we will not support thrift.

Reply:
Closing this since it seems to be agreed that we are moving to disable Thrift completely. I'll comment on what that means in terms of additional changes below.